pkg_resources/__init__.py

```diff
@@ -2477,11 +2477,26 @@ def _dep_map(self):
                             extra, marker = extra.split(':', 1)
                             if invalid_marker(marker):
                                 # XXX warn
-                                reqs=[]
+                                reqs = []
                             elif not evaluate_marker(marker):
-                                reqs=[]
+                                reqs = []
                         extra = safe_extra(extra) or None
-                    dm.setdefault(extra,[]).extend(parse_requirements(reqs))
+                    try:
+                        dm.setdefault(extra, []).extend(
```

Comment:
Probably better not to trap exceptions for dm.setdefault().extend. maybe split this clause into two clauses `parsed = parsed_requirements(reqs); dm.setdefault(extra, []).extend(parsed)` and then only trap the exception for the call that's expected to raise the exception.